gpm_api/tests/conftest.py

```diff
@@ -0,0 +1,322 @@
+import posixpath as pxp
+import ntpath as ntp
+import gpm_api.configs
+from unittest.mock import patch
+
+
+@pytest.fixture(scope="session", autouse=True)
+def mock_configuration():
+    """Patch the user configuration for entire session
+
+    Doing this will retrieve the configuration from pytest memory and not
+    alter the local configuration in ~/.config_gpm_api.yml
+    """
+
+    mocked_configuration = {
+        "gpm_username": "testuser",
+        "gpm_password": "testuser",
+        "gpm_base_dir": "data",
+    }
+
+    with patch.object(
+        gpm_api.configs,
+        "read_gpm_api_configs",
+        return_value=mocked_configuration,
+    ):
+        yield
+
+
+@pytest.fixture
+def product_types() -> List[str]:
+    """Return a list of all product types from the info dict"""
+    product_types = []
+    for product, props in get_info_dict().items():
+        product_types += props["product_types"]
+
+    product_types = list(set(product_types))  # Dedup list
+
+    return product_types
+
+
+@pytest.fixture
+def product_categories() -> List[str]:
+    """Return a list of product categories from the info dict"""
+
+    return list(set([props["product_category"] for props in get_info_dict().values()]))
+
+
+@pytest.fixture
+def product_levels() -> List[str]:
+    """Return a list of product levels from the info dict"""
+
+    # Available in gpm_api.io.checks.check_product_level()
+    return ["1A", "1B", "1C", "2A", "2B"]
+
+
+@pytest.fixture
+def versions() -> List[int]:
+    """Return a list of versions"""
+
+    # Available in gpm_api.io.checks.check_version()
+    return [4, 5, 6, 7]
+
+
+@pytest.fixture
+def products() -> List[str]:
+    """Return a list of all products regardless of type"""
+
+    return available_products()
+
+
+@pytest.fixture
+def username() -> str:
+    """Return a username
+
+    GPM uses an email address as username
+    """
+
+    return "testuser@example.com"
+
+
+@pytest.fixture
+def password() -> str:
+    """Return a password
+
+    GPM password is the username
+    """
+
+    return "testuser@example.com"
+
+
+@pytest.fixture
+def server_paths() -> Dict[str, Dict[str, Any]]:
+    """Return a list of probable GPM server paths"""
+
+    # Not validated to be real paths but follow the structure
+    return {
+        "ftps://arthurhouftps.pps.eosdis.nasa.gov/gpmdata/2020/07/05/radar/2A.GPM.DPR.V9-20211125.20200705-S170044-E183317.036092.V07A.HDF5": {
+            "year": 2020,
+            "month": 7,
+            "day": 5,
+            "product": "2A-DPR",
+            "product_category": "radar",
+            "product_type": "RS",
+            "start_time": datetime.datetime(2020, 7, 5, 17, 0, 44),
+            "end_time": datetime.datetime(2020, 7, 5, 18, 33, 17),
+            "version": 7,
+        },
+        "ftps://arthurhouftps.pps.eosdis.nasa.gov/gpmdata/2020/07/05/radar/2A.GPM.DPR.V9-20211125.20200705-S183318-E200550.036093.V07A.HDF5": {
```

Comment:
Why here you repeat similar paths ?

Reply:
To satisfy the need to try different criteria in other test functions I worked with one path and modified it slightly to assert differences in properties. As mentioned above, the exact structure and properties have not been completely understood when writing the functions and some assistance in defining real paths would be helpful. 

In any case, I imagine that when updated, the tests should adapt to those without any trouble.